libnavui-dropin/src/main/java/com/mapbox/navigation/dropin/NavigationViewApi.kt

```diff
@@ -19,45 +26,86 @@ class NavigationViewApi internal constructor(
 ) {
 
     /**
-     * The action is used to request and set routes based on [RouteOptions].
-     * @param options
+     * Request routes based on [RouteOptions]. Routes obtained from this route request can be
+     * obtained via
+     * [NavigationViewListener.onRouteFetching] when the route is being fetched
+     * [NavigationViewListener.onRouteFetchFailed] if the route request failed
+     * [NavigationViewListener.onRouteFetchCanceled] if the route request was canceled
+     * [NavigationViewListener.onRouteFetchSuccessful] if the route request was a success
+     *
+     * @param options [RouteOptions]
      */
-    fun setRoutes(options: RouteOptions) {
-        store.dispatch(RoutesAction.FetchOptions(options))
+    fun fetchRoutes(options: RouteOptions) {
+        store.dispatch(RoutePreviewAction.FetchOptions(options))
     }
 
     /**
-     * The action is used to directly set the [NavigationRoute] supplied to [NavigationView].
+     * Request routes based on list of [Point]. Routes obtained from this route request can be
+     * obtained via
+     * [NavigationViewListener.onRouteFetching] when the route is being fetched
+     * [NavigationViewListener.onRouteFetchFailed] if the route request failed
+     * [NavigationViewListener.onRouteFetchCanceled] if the route request was canceled
+     * [NavigationViewListener.onRouteFetchSuccessful] if the route request was a success
+     * The first point in the list should be the origin and last point should be the destination.
+     * If you have multi waypoints, you can insert maximum of 23 additional waypoints in the list
+     * between the first and the last point, resulting in a total maximum of 25 points.
+     *
+     * @param points list of [Point]
+     */
+    @Throws
+    fun fetchRoutes(points: List<Point>) {
+        if (points.size > 25) {
```

Comment:
Is this needed? seems like we can remove this and have it be handled by `onRouteFetchFailed`

Reply:
I would prefer it to fail fast rather than make a request only to return an error.

Reply:
I wanted it to fail fast so that there is no request at all if we know it will always fail. But, on the other hand, we wouldn't know upstream about this limit if it changes on Nav native side since native makes the route request. In order to maintain consistency and avoid regressions in the future I am gonna remove it.